### PR TITLE
signature helps are not shown if the target code has comments

### PR DESCRIPTION
**Pull request by tk0miya**
SignatureHelpProvider fails to detect the cursor position correctly because it uses source code from AST node instead of original source code.  Therefore it misdetects the cursor position if the source code contains comments.  As a result, it does not show signature helps it the source code contains comments.

### CODE REVIEW

File: lib/steep/services/signature_help_provider.rb [hunk 1]
```diff
@@ -11,15 +11,10 @@ def env
         subtyping.factory.env
       end
 
-      def initialize(source:, subtyping:)
+      def initialize(source:, content:, subtyping:)
```
Comment (tk0miya) [inline lib/steep/services/signature_help_provider.rb:14]: I don't know adding a new argument here is a better way. It seems also better to add a `content` attribute to `Steep::Source` object. Please let me know your opinion.

File: lib/steep/services/signature_help_provider.rb [hunk 2]
```diff
@@ -11,15 +11,10 @@ def env
...
         @source = source
         @subtyping = subtyping
-
-        text =
-          if source.node
-            source.node.loc.expression.source
-          end
-        @buffer = RBS::Buffer.new(name: source.path, content: text || "")
+        @buffer = RBS::Buffer.new(name: source.path, content: content)
```
Comment (soutaro) [inline lib/steep/services/signature_help_provider.rb:17]: It looks like we can simply use `Source#buffer` here.

```suggestion
        @buffer = source.buffer
```

**Review by soutaro — CHANGES_REQUESTED** [review]
Thank you for the fix. It makes sense!

How about simply using the `source.buffer`? If it works, I also want to ask you to add a test case that has comments before the Ruby code.

File: lib/steep/services/signature_help_provider.rb (hunk 2 above)
Comment (soutaro) [reply lib/steep/services/signature_help_provider.rb:17]: `Source#buffer` was added very recently, weeks ago.

Comment (tk0miya) [reply lib/steep/services/signature_help_provider.rb]: Great! It works fine!

**Comment by tk0miya** [comment]
Thank you for your valuable comment. I updated this PR :-)

**Review by soutaro — APPROVED** [review]
🎉

**Pull request merged**